integration_test/dr-buckets.tf.template

```diff
@@ -1,17 +1,58 @@
 resource "aws_s3_bucket" "orca-archive-worm" {
   bucket = "PREFIX-orca-archive-worm"
   force_destroy = true
+  object_lock_enabled = true
   tags = {
     Deployment = "PREFIX"
     Application = "ORCA"
   }
 }
+
 resource "aws_s3_bucket_versioning" "orca-archive-worm-versioning" {
   bucket = aws_s3_bucket.orca-archive-worm.id
   versioning_configuration {
     status = "Enabled"
   }
 }
+
+resource "aws_s3_bucket_object_lock_configuration" "orca-archive-worm-object-lock-config" {
+  bucket = aws_s3_bucket.orca-archive-worm.bucket
+  rule {
+    default_retention {
+      mode = "GOVERNANCE"
+      days = 180
```

Comment:
kept it as 6 months according to @AndrewDorn suggestion in his research doc:
"With a default policy of Governance mode and an arbitrarily long Retention period, data will be protected from accidental deletion by general users."